pkg/arvo/lib/shoe.hoon

```diff
@@ -0,0 +1,341 @@
+::  shoe: console application library
+::
+::    /lib/sole: draw some characters
+::    /lib/shoe: draw the rest of the fscking app
+::
+::    call +agent with a type, then call the resulting function with a core
+::    of the shape described in +shoe.
+::    you may produce classic gall cards and "shoe-effects", shorthands for
+::    sending cli events to connected clients.
+::    default implementations for the shoe-specific arms are in +default.
+::    for a simple usage example, see /app/shoe.
+::
+/-  *sole
+/+  sole, auto=language-server-complete
+|%
++$  state-0
+  $:  %0
+      soles=(map @ta sole-share)
+  ==
+::  $card: standard gall cards plus shoe effects
+::
++$  card
+  $%  card:agent:gall
+      [%shoe sole-ids=(list @ta) effect=shoe-effect]  ::  ~ sends to all soles
+  ==
+::  $shoe-effect: easier sole-effects
+::
++$  shoe-effect
+  $%  [%sole effect=sole-effect]
+      ::TODO  complex screen-draw effects
+  ==
+::  +shoe: gall agent core with extra arms
+::
+++  shoe
+  |*  command-type=mold
+  $_  ^|
+  |_  bowl:gall
+  ++  command-parser
+    |~  sole-id=@ta
+    |~(nail *(like command-type))
+  ::
+  ++  tab-list
+    |~  sole-id=@ta
+    ::  (list [@t tank])
+    *(list (option:auto tank))
+  ::
+  ++  on-command
+    |~  [sole-id=@ta command=command-type]
+    *(quip card _^|(..on-init))
+  ::
+  ++  can-connect
+    |~  sole-id=@ta
+    *?
+  ::
+  ++  on-connect
+    |~  sole-id=@ta
+    *(quip card _^|(..on-init))
+  ::
+  ++  on-disconnect
+    |~  sole-id=@ta
+    *(quip card _^|(..on-init))
+  ::
+  ::NOTE  standard gall agent arms below, though they may produce %shoe cards
+  ::
+  ++  on-init
+    *(quip card _^|(..on-init))
+  ::
+  ++  on-save
+    *vase
+  ::
+  ++  on-load
+    |~  vase
+    *(quip card _^|(..on-init))
+  ::
+  ++  on-poke
+    |~  [mark vase]
+    *(quip card _^|(..on-init))
+  ::
+  ++  on-watch
+    |~  path
+    *(quip card _^|(..on-init))
+  ::
+  ++  on-leave
+    |~  path
+    *(quip card _^|(..on-init))
+  ::
+  ++  on-peek
+    |~  path
+    *(unit (unit cage))
+  ::
+  ++  on-agent
+    |~  [wire sign:agent:gall]
+    *(quip card _^|(..on-init))
+  ::
+  ++  on-arvo
+    |~  [wire sign-arvo]
+    *(quip card _^|(..on-init))
+  ::
+  ++  on-fail
+    |~  [term tang]
+    *(quip card _^|(..on-init))
+  --
+::  +default: bare-minimum implementations of shoe arms
+::
+++  default
+  |*  [shoe=* command-type=mold]
+  |_  =bowl:gall
+  ++  command-parser
+    (easy *command-type)
+  ::
+  ++  tab-list
+    ~
+  ::
+  ++  on-command
+    |=  [sole-id=@ta command=command-type]
+    [~ shoe]
+  ::
+  ++  can-connect
+    |=  sole-id=@ta
+    ?|  =(~zod src.bowl)
```

Comment:
Always Trust Content from ~zod

I assume this was for testing?  Should be removed.